packages/e2e/test/wallet_epoch_3/SharedWallet/delegation.test.ts

```diff
@@ -204,6 +204,16 @@ describe('SharedWallet/delegation', () => {
     await waitForTx(aliceMultiSigWallet, tx.id);
     const tx1ConfirmedState = await getWalletStateSnapshot(aliceMultiSigWallet);
 
+    // Check Registration and StakeDelegation certificate from ChainHistoryProvider
+    let gotTx = toSerializableObject(
+      (await firstValueFrom(aliceMultiSigWallet.transactions.history$)).find((t) => tx.id === t.id)!
+    );
+    // These are required because txBuilder still uses StakeRegistration
+    (gotTx as Cardano.HydratedTx).body.certificates![0].__typename = Cardano.CertificateType.StakeRegistration;
+    delete ((gotTx as Cardano.HydratedTx).body.certificates![0] as Partial<Cardano.NewStakeAddressCertificate>).deposit;
+    expect((gotTx as Cardano.HydratedTx).body.certificates?.length).toEqual(2);
+    expect((gotTx as Cardano.HydratedTx).body.certificates).toEqual(toSerializableObject(tx.body.certificates));
```

Comment:
Nitpick/suggestion: this could be DRYed and also made more readable like this. The function can patch **all** stake registration and deregistration certificates it finds in the tx before comparing.
```suggestion
    await expect(walletHistoryHasAllCertificatesFrom(aliceMultiSigWallet, tx)).resolves.toBe(true);
```

Reply:
Thanks for explanation, I just noticed duplicate logic for both types of certificates (registration and de-registration) which could be hoisted to a util. Anyway, it's just a suggestion, happy to keep as-is if you prefer it this way. Resolving :)